escrow/test/Escrow.js

```diff
@@ -0,0 +1,106 @@
+
+  const ESCROW_AMOUNT = 1_000_000_000; // 0.001 ACA
+
+  beforeEach(async function () {
+    [deployer, requestor, provider] = await ethers.getSigners();
+
+    requestorAddress = await requestor.getAddress();
+    providerAddress = await provider.getAddress();
+
+    acaInstance = new Contract(ACA, TokenContract.abi, requestor);
+
+    Escrow = new ContractFactory(EscrowContract.abi, EscrowContract.bytecode, deployer);
+    escrowInstance = await Escrow.deploy(acaInstance.address, ESCROW_AMOUNT, requestorAddress, providerAddress,
+      {
+        gasPrice: ethParams.txGasPrice,
+        gasLimit: ethParams.txGasLimit,
+      }
+    );
+  });
+
+  it("deployment should assign the escrow provider and requestor", async function () {
+    expect(await escrowInstance.requestor()).to.equal(requestorAddress);
+    expect(await escrowInstance.serviceProvider()).to.equal(providerAddress);
+  });
+
+  it("should only allow the requestor to complete his part", async function () {
+    await expect(escrowInstance.connect(provider).requestorConfirmTaskCompletion(true)).to.
+      be.revertedWith("Only the requestor can confirm his part")
+  });
+
+  it("should only allow the service provider to complete his part", async function () {
+    await expect(escrowInstance.connect(requestor).serviceProviderConfirmTaskCompletion(true)).to.
+      be.revertedWith("Only the service provider can confirm his part")
+  });
+
+  it("should let the both parties confirm the task completion and send tokens to the service provider", async function () {
+    // transfer funds to the contract
+    const previousContractBalance = await acaInstance.balanceOf(escrowInstance.address);
+    await acaInstance.transfer(escrowInstance.address, ESCROW_AMOUNT, { from: requestorAddress })
+    const currentContractBalance = await acaInstance.balanceOf(escrowInstance.address);
+    expect(currentContractBalance).to.equal(previousContractBalance.add(ESCROW_AMOUNT));
+
+    // requestor and service provider confirm completion
+    await escrowInstance.connect(provider).serviceProviderConfirmTaskCompletion(true)
+    const initialProviderNativeBalance = await provider.getBalance();
+    const initialProviderBalance = await acaInstance.balanceOf(providerAddress);
+    await escrowInstance.connect(requestor).requestorConfirmTaskCompletion(true)
+
+    // complete the task and check if the provider balance increased
+    const finalProviderBalance = await acaInstance.balanceOf(providerAddress);
+    expect(finalProviderBalance).to.equal(initialProviderBalance.add(ESCROW_AMOUNT));
+
+    // the change in native ACA balance should match the change in ERC20 ACA balance
+    const finalProviderNativeBalance = await provider.getBalance();
+    // multiplied the escrow amount by 10⁶ because there's a difference in
+    // BigNumber's decimals between native and ERC20 balance
+    const expectedProviderNativeBalance = initialProviderNativeBalance.add(ESCROW_AMOUNT * 1_000_000);
```

Comment:
This is a weird workaround I had to do because the `BigNumber` returned by the native balance is different than the `BigNumber` returned by the `balanceOf` of the `acaInstance`. The former preserves all the zeroes while the latter trims them out. So I had to multiply it by 10^6 for them to have the same amount of decimals.

@ThunderDeliverer any suggestions how to handle this in a better way?

Reply:
In stead of doing the multiplication, you can just check both in the same way (either both with `getBalance` or by querying `balanceOf` from the `ERC20` instance). I'd suggest the second option